app/controllers/queue_items_controller.rb

```diff
@@ -0,0 +1,28 @@
+class QueueItemsController < ApplicationController
+  before_filter :require_user
+  
+  def index
+    @queue_items = current_user.queue_items
+  end
+  
+  def create
+    video = Video.find(params[:video_id])
+    QueueItem.create(video: video, user: current_user, position: new_queue_item_position) unless current_user_queued_video?(video)
+    redirect_to my_queue_path
+  end
+  
+  def destroy
+    queue_item = QueueItem.find(params[:id])
+    queue_item.destroy if queue_item.user == current_user
+    redirect_to my_queue_path
+  end
+  
+  private
+    def new_queue_item_position
+      current_user.queue_items.count + 1
+    end
+
+    def current_user_queued_video?(video)
```

Comment:
Since this method is attached to a user object it could easily be passed and refactored to `User` model. Also naming it as `current_user_` should be a sign for you to push such behavior to appropriate place.